spec/models/payload_spec.rb

```diff
@@ -0,0 +1,103 @@
+require_relative '../spec_helper.rb'
+require 'pry'
+
+RSpec.describe "Payload" do
+  it "is invalid without a url" do
+
+    result = create_payload_hash({url_id: nil})
+
+    expect(Payload.create(result)).to_not be_valid
```

Comment:
we can use `new` instead of `create` here to avoid unnecessary saving of data in our database. we just care if it's able to be saved in our database (aka it's valid) rather than the actual saving.